File: factorial/factorial.py
def get_factorial(number) -> int:
     assert number >= 0
     if number < 2:
          return 1
     else:
          return get_factorial(number - 1) * number


print(f"Factorial of 5: {get_factorial(5)}")
print(f"Factorial of 10: {get_factorial(10)}")
print(f"Factorial of 0: {get_factorial(0)}")

File: factorial/factorial.go
package main

import "fmt"

func main() {
	fmt.Printf("Factorial of 5: %d\n", getFactorial(5))
	fmt.Printf("Factorial of 10: %d\n", getFactorial(10))
	fmt.Printf("Factorial of 0: %d\n", getFactorial(0))
}

func getFactorial(number uint) uint {
	if number < 2 {
		return 1
	}
	return getFactorial(number-1) * number
}
